### PR TITLE
Add brief , default and html summary NUnit 3 transforms

### PR DESCRIPTION
**Pull request by harrisonmeister**
Add brief and default transforms compatible with nunit v3 xml output reports. These try to match the Nunit3 console output as closely as possible.

This is the first 2 of the 4 transforms for #5 

@CharliePoole / @rprouse - as this is my first PR into the nunit.org, I wasn't sure if there was any additional steps I need to take for this, so please feel free to point me in the right direction if there is!

The brief output provides a summary like this:
[brief-v3.txt](https://github.com/nunit/nunit-summary/files/461080/brief-v3.txt)

``` txt
Runtime Environment 
   OS Version: Microsoft Windows NT 10.0.10586.0
  CLR Version: 4.0.30319.42000

NUnit Version: 3.4.1.0

Test Run Summary 
 Overall result: Failed
 Test Count: 14, Passed: 2, Failed: 6, Inconclusive: 2, Skipped: 4
   Failed Tests - Failures: 2, Errors: 2, Invalid: 2
   Skipped Tests - Ignored: 2, Explicit: 2, Other: 0
 Start time: 2016-09-08 07:40:49Z
   End time: 2016-09-08 07:40:49Z
   Duration: 0.146 seconds
```

whilst the Default provides a summary like so:
[default-v3.txt](https://github.com/nunit/nunit-summary/files/461079/default-v3.txt)

``` txt
Runtime Environment 
   OS Version: Microsoft Windows NT 10.0.10586.0
  CLR Version: 4.0.30319.42000

NUnit Version: 3.4.1.0

Test Files 
    C:\Users\mark.harrison\Documents\visual studio 2015\Projects\ClassLibrary1\bin\Debug\ClassLibrary1.dll
    C:\Users\mark.harrison\Documents\visual studio 2015\Projects\ClassLibrary1\bin\Debug\ClassLibrary1.dll

Tests Not Run

1) Explicit : ClassLibrary1.ExplicitTests.The_TEST
Some Explicit reason

2) Ignored : ClassLibrary1.NormalTests.IgnoreMe
Some Ignore reason

3) Explicit : ClassLibrary1.ExplicitTests.The_TEST
Some Explicit reason

4) Ignored : ClassLibrary1.NormalTests.IgnoreMe
Some Ignore reason

Errors and Failures

1) Invalid : ClassLibrary1.BadFixture.SomeTest
OneTimeSetUp: No suitable constructor was found

2) Error : ClassLibrary1.NormalTests.Error_Case
System.ApplicationException : error happened
   at ClassLibrary1.NormalTests.Error_Case() in C:\Users\mark.harrison\Documents\Visual Studio 2015\Projects\ClassLibrary1\Class1.cs:line 46

3) Failed : ClassLibrary1.NormalTests.Failure_Case
Intentional Failure
at ClassLibrary1.NormalTests.Failure_Case() in C:\Users\mark.harrison\Documents\Visual Studio 2015\Projects\ClassLibrary1\Class1.cs:line 40

4) Invalid : ClassLibrary1.BadFixture.SomeTest
OneTimeSetUp: No suitable constructor was found

5) Error : ClassLibrary1.NormalTests.Error_Case
System.ApplicationException : error happened
   at ClassLibrary1.NormalTests.Error_Case() in C:\Users\mark.harrison\Documents\Visual Studio 2015\Projects\ClassLibrary1\Class1.cs:line 46

6) Failed : ClassLibrary1.NormalTests.Failure_Case
Intentional Failure
at ClassLibrary1.NormalTests.Failure_Case() in C:\Users\mark.harrison\Documents\Visual Studio 2015\Projects\ClassLibrary1\Class1.cs:line 40

Run Settings
    DefaultTimeout: 900000
    WorkDirectory: D:\TestResults
    ImageRuntimeVersion: 4.0.30319
    ImageTargetFrameworkName: .NETFramework,Version=v4.5
    ImageRequiresX86: False
    ImageRequiresDefaultAppDomainAssemblyResolver: False
    NumberOfTestWorkers: 8

Test Run Summary 
 Overall result: Failed
 Test Count: 14, Passed: 2, Failed: 6, Inconclusive: 2, Skipped: 4
   Failed Tests - Failures: 2, Errors: 2, Invalid: 2
   Skipped Tests - Ignored: 2, Explicit: 2, Other: 0
 Start time: 2016-09-08 07:40:49Z
   End time: 2016-09-08 07:40:49Z
   Duration: 0.146 seconds
```

Also, here is the nunit3 result output in case its useful for comparison
[testing-nunit3.zip](https://github.com/nunit/nunit-summary/files/461084/testing-nunit3.zip)

### CODE REVIEW

**Comment by rprouse** [comment]
@harrisonmeister thanks for the contribution. :+1: 

Before I review, you asked about how to do an NUnit PR. What you submitted is fine, but I made some edits to make your next PR easier.
1. You don't need to use the title that GitHub suggests, especially if it ends up split. Edit it to make it short and concise so we can quickly understand it when looking at the PR list.
2. You referenced the issue, that is great, it creates cross links
3. If a PR fixes an issue, add Fixes #nnn to the PR so the issue will be automatically closed. Not needed here.
4. I copied the results inline to make it easier for reviewers. Downloading and opening files slows us down and we often do quick reviews on our mobile devices which is difficult with downloads.

Now, to the review... :smile:

**Comment by harrisonmeister** [comment]
Thanks for the feedback @rprouse - will keep it in mind for the next PR

**Comment by rprouse** [comment]
The code changes look good to me. It would be nice to add unit tests soon before there are too many transforms to catch up on, but I am okay with finishing these four transforms first. I also need to download this project and familiarize myself with the code and features :smile: 

I don't like that the brief summary does not show the errors, but I assume that it is functionally equivalent to the NUnit 2 version?

@CharliePoole can you do a second review.

**Comment by harrisonmeister** [comment]
@rprouse - correct, the v2 xslt didnt include details of tests which errored, but I can add that to the v3 ones (its essentially just the same info from the default v3 one) so let me know if you want that added too.

**Comment by rprouse** [comment]
I also assume that detecting the transform to use based on the v2/v3 output will be done later? Looking at the code, I see that the transforms are hard coded.

**Comment by rprouse** [comment]
I also wonder if we should rename the existing transforms to `-v2` (and update the code) to make them consistent?

**Comment by CharliePoole** [comment]
I made auto detection a separate issue because it requires a bit of reorganizing in the code. Currently the transform to use is based only on the options so it is constructed once before any input is read. It will have to be done now for each input file.

BTW, the list of issues I created serves as a sort of plan of attack.

**Comment by CharliePoole** [comment]
@harrisonmeister One addition to @rprouse's advice for doing PRs...

Since you have write access to the repository, you should create the branches there. That gets you a build just by pushing the branch, even before you create a PR and also makes it easier for collaboration when it's needed. Our practice is still to use a PR and get another committer to approve the merge before doing it.

If you should end up working on any other nunit projects, like the framework or console runner, then you will be back to using your fork at least initially but because of your interest in the project I gave you write access to nunit-summary from the start.

**Comment by CharliePoole** [comment]
I had some minor questions about format, but I think it will be better to do a final review after all eight transforms are in place and the autoselect bit is done.

File: src/XmlTransformer.cs [hunk 1]
```diff
@@ -109,9 +109,6 @@ static void Main(string[] args)
 
                 if (!options.MultipleOutput)
                 {
-                    if ( options.Html && !options.NoHeader )
```
Comment (CharliePoole) [inline src/XmlTransformer.cs:112]: It turns out this is incorrect. It leads to multiple  trailers when all the transforms are going to the same file. It was the first instance of WriteHtmlTrailer that needed to be suppressed in this case.

I'm going to restructure the code to take care of this under issue #6. I think there should just be two different methods for doing output: one for merging to a single file and another to doing one output per input.

**Comment by harrisonmeister** [comment]
Ah ok, thanks @CharliePoole for the review. I knew it was one of them :) 
- And for the tip on the branches direct in the summary repo!

**Pull request merged**